src/Weather/Skysight/Metrics.hpp

```diff
@@ -24,6 +24,71 @@ Copyright_License {
 #ifndef WEATHER_SKYSIGHT_METRICS_HPP
 #define WEATHER_SKYSIGHT_METRICS_HPP
 
+#include "Util/tstring.hpp"
+#include "Time/BrokenDateTime.hpp"
 #include <map>
 
+struct LegendColor {
+  unsigned char Red;
+  unsigned char Green;
+  unsigned char Blue;
+};
+
+struct SkysightMetric {
+  const tstring id;
+  const tstring name;
+  const tstring desc;
+  uint64_t last_update = 0;
+  std::map<float, LegendColor> legend;
```

Comment:
Eh, using `float` as a map key - that's a WTF!

Reply:
I dump the key when legend is created. It is 3.5, 2400, -600, 1.6, -3.5, ... It is really floating point. What is your point? Key should not be floating point?

Reply:
No, keys should not be floating point. Maybe a `map` is the wrong data structure here. I don't know, because you forgot to add documentation here, so nobody knows what you're trying to do.